### ISSUE TITLE
FP / Discussion: Domain can be compromised with simple variable assignment

### ISSUE THREAD

**Issue by andreasschroth** [kalessil/phpinspectionsea#1350]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Discussion / False-Positive                          |
| Plugin         | Php Inspections (EA Ultimate), 3.0.1 |
| Language level | PHP 7.3                        |

Current behaviour:
---

Here is some example code to illustrate the case:

```php
<?php declare(strict_types = 1);

use Foo\ErrorHandler;
use Foo\ExceptionHandler;

$host = $_SERVER['HTTP_HOST'] ?? null;
\set_error_handler([new ErrorHandler($config, $logger, $host), 'log']);
\set_exception_handler([new ExceptionHandler($logger, $host), 'log']);
```

It triggers the warning "The domain here can be compromised, consider introducing whitelists.".

Suggested behaviour:
---

I definitely understand the intention behind it (do not rely on user-input, also the given server host could be compromised), but in this case, no comparison / whatever is done with the hostname. It's just passed to the error / exception handlers in this example which then do nothing.

Please feel free to close the issue if you think this code is a valid case of the warning being triggered. I'm not absolutely sure myself as it's difficult to analyse only the valid use-cases.

**Comment by kalessil** [comment]
In this case, renaming the variable to something not having `domain`, `email`, `host` will make the trick. 
E.g. `node`, `instance`.

**Comment by kalessil** [comment]
Apart from the tweak, the warning seems to be legit - the possible substituted hostname is dispatched to a call, hence we could be compromised.

**Comment by andreasschroth** [comment]
@kalessil Thanks! Wasn't aware it's that easy to bypass the problem (renaming the variable).
